src/orders-update.js

```diff
@@ -1,26 +1,85 @@
-import { SphereClient } from 'sphere-node-sdk'
+import Ajv from 'ajv'
+import Promise from 'bluebird'
+import { SphereClient, OrderSync } from 'sphere-node-sdk'
 
-export default class OrdersUpdate {
+import orderSchema from './order-schema'
 
-  constructor (apiClientConfig, config = {}, logger) {
-    this.client = new SphereClient(
-      apiClientConfig
-      // Object.assign(apiClientConfig, { user_agent: userAgent })
-    )
+const ajv = new Ajv({ removeAdditional: true })
 
-    this.config = config
+export default class OrdersUpdate {
 
-    this.logger = logger || {
-      error: process.stderr,
-      warn: process.stderr,
-      info: process.stdout,
-      verbose: process.stdout,
-    }
+  constructor (apiClientConfig) {
+    this.client = new SphereClient(apiClientConfig)
 
     this.summary = {
       errors: [],
       inserted: [],
       successfullImports: 0,
     }
   }
+
+  summaryReport () {
+    return JSON.stringify(this.summary, null, 2)
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  validateOrderData (order) {
+    const validatedOrderData = ajv.compile(orderSchema)(order)
+
+    if (validatedOrderData)
+      return Promise.resolve(order)
+
+    return Promise.reject(`Validation error: ${validatedOrderData.errors}`)
+  }
+
+  processOrder (order) {
+    return this.validateOrderData(order)
+      .then(this.updateOrder.bind(this))
```

Comment:
Don't you think binding is more explicit? Feel like using an arrow function is implicit.